presto-main/src/main/java/io/prestosql/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -885,14 +885,20 @@ private static InnerJoinPushDownResult processInnerJoin(
             ImmutableSet<Symbol> leftScope = ImmutableSet.copyOf(leftSymbols);
             ImmutableSet<Symbol> rightScope = ImmutableSet.copyOf(rightSymbols);
 
+            // Attempt to simplify the effective left/right predicates with the predicate we're pushing down
+            // This, effectively, inlines any constants derived from such predicate
+            EqualityInference predicateInference = EqualityInference.newInstance(metadata, inheritedPredicate);
```

Comment:
Could we build more nested plan, where there are still not enough predicate pushdowns iterations?
See: https://github.com/prestodb/presto/issues/11569 (inference for outer joins might is limited and predicates might be derived only after n-th run depending how nested joins are)